.github/workflows/check-catalyst.yaml

```diff
@@ -265,6 +265,7 @@ jobs:
         key: ${{ runner.os }}-llvm-${{ steps.llvm-hash.outputs.llvm-hash }}-default-build
         fail-on-cache-miss: True
 
+    # TODO: it doesn't have any use case here! Remove it?
     - name: Cache CCache
```

Comment:
I think we can remove this one here.

Reply:
This is the ccache cache. If we are using ccache, we need access to the cache, don't we? Removing it should remove the cache unless I am mistaken.

Reply:
Yeah! That was my question?! I cannot see where this cache is used in the action and wanna know if everyone is against removing it.

Reply:
We shouldn't remove the cache, because otherwise, every time we build we won't have a cache. I believe that this is actually using just setting the cache folder and re-using previous caches and ccache will use it since it is in the /home/user/.cache location. I might be wrong, will investigate a bit.

Reply:
Yes this cache is very important for the quantum build and we do use it to speed up the action.

Reply:
Oops! That's my misunderstanding of `actions/cache@v3` :sweat_smile: Thanks for the clarification btw! :)